src/hopen/syntax/mustache.cljc

```diff
@@ -0,0 +1,434 @@
+
+;;------------------------------------------------------------------------------
+;; General utilities
+;;------------------------------------------------------------------------------
+
+(defn re-seq-pos
+  "Returns a lazy sequence of regex matches like `re-seq`, but also returns the offsets of
+  the groups/captures."
+  [pattern string]
+  #?(:clj
+     (let [m (re-matcher pattern string)]
+       ((fn step []
+          (when (. m find)
+            (cons {:start (.start m)
+                   :end (.end m)
+                   :groups (into [] (for [g (range (inc (.groupCount m)))]
+                                      (.group m g)))}
+                  (lazy-seq (step)))))))
+
+     :cljs
+     ;; It'd be really nice if we can just use the global flag for the js regex
+     ;; However, https://dev.clojure.org/jira/browse/CLJS-150 (from 2014) suggets that
+     ;; it makes use of some global state, which may mess up the result.
+     ;;
+     ;; This is an altered version of cljs' re-seq.
+     (letfn [(re-seq-pos* [re s current-offset]
+               (when-some [matches (.exec re s)]
+                 (let [match-str (aget matches 0)
+                       start (+ current-offset (.-index matches))
+                       end (+ start (count match-str))]
+                   (cons {:start start
+                          :end end
+                          :groups (if (== (.-length matches) 1)
+                                    match-str
+                                    (vec matches))}
+                         (lazy-seq
+                          (let [post-idx (+ (.-index matches)
+                                            (max 1 (.-length match-str)))]
+                            (when (<= post-idx (.-length s))
+                              (re-seq-pos* re (subs s post-idx) (+ current-offset post-idx)))))))))]
+     (if (string? string)
+         (re-seq-pos* pattern string 0)
+         (throw (js/TypeError. "re-seq-pos must match against a string."))))))
+
+(defn conj-last
+  "Conj `x` into the last element of the `coll`.
+  `coll` should be a seq of seqs."
+  [coll x]
+  (conj (pop coll) (conj (peek coll) x)))
+
+(defn conj-some
+  [coll & xs]
+  (->> xs
+       (filter some?)
+       (apply conj coll)))
+
+(defn deep-merge
+  [a b]
+  (if (map? a)
+    (into a (for [[k v] b]
+              [k (deep-merge (a k) v)]))
+    b))
+
+;;------------------------------------------------------------------------------
+;; Namespace specific functions
+;;------------------------------------------------------------------------------
+
+(defn- compile-regex [start-delim end-delim]
+  (if (and (= start-delim "{{")
+           (= end-delim "}}"))
+    #"\{\{\{(.*?)\}\}\}|\{\{(.*?)\}\}"
+    (re-pattern
+     (str
+      (util/re-quote start-delim)
+      "\\{(.*?)\\}"
+      (util/re-quote start-delim)
+      "|"
+      (util/re-quote start-delim)
+      "(.*?)"
+      (util/re-quote end-delim)))))
+
+(defn- parse-delim-change-tag
```

Comment:
```clojure
(parse-delim-change-tag "=foo bar fizz=")
=> ("foo bar" "fizz")
```

Reply:
What do you think it should do in this case? One way to handle this is just to throw an exception, I suppose.

Reply:
I wanted to emphasize on `.*` matching too many things. I propose to use `\\S*` instead.
I have no opinion on the error management.

Reply:
done